### PR TITLE
MAINT: Clarify policy language in NEP-29.

### PR DESCRIPTION
**Pull request by stefanv**
This is to clarify and make consistent some of the language in NEP 29.

@tacaswell I need some feedback from you on the section marked "XXX Tom".  It concerns the situation where a project releases directly after a *major* Python release.

### CODE REVIEW

**Comment by mattip** [comment]
There is currently a proposal [PEP 602](https://discuss.python.org/t/pep-602-annual-release-cycle-for-python/) to increase the rlease cadence of CPython from 18 months to 12 months. This means projects will commit to fully supporting 4 (well, 3-4) versions of python, rather than 2 (2-3) under the old cadence.

File: doc/neps/nep-0029-deprecation_policy.rst [hunk 1]
```diff
@@ -65,33 +65,32 @@ The diagram::
...
+interval for Python to provides resilience against small fluctuations /
+delays in its release schedule.
+
+Because Python minor version support is based only on historical
+release dates, a 42 month time window, and a planned project release
+date, one can predict with high confidence when a project will be able
+to drop any given minor version of Python.  This, in turn, could save
+months of unnecessary maintenance burden.
+
+If a project releases immediately after a minor version of Python
```
Comment (stefanv) [inline doc/neps/nep-0029-deprecation_policy.rst]: @tacaswell I am now happy with the wording of the whole NEP, except for these two paragraphs, which I am still not so sure about.  It seems to me as though there are *two* situations that create discrepancies: a release shortly after a minor version drops out of support, or shortly after a new minor version enters.  In either case, the project will support a different subsets of Python.  Subsequent releases of those packages will fix it up, but I think the word "guaranteed" below is then too strong a statement.  We cannot guarantee when packages release, so if they take a long time to release after a new version of Python is introduced we're in trouble anyway.

**Comment by tacaswell** [comment]
@mattip I think that is ok and within the intent of this NEP.

File: doc/neps/nep-0029-deprecation_policy.rst (hunk 1 above)
Comment (tacaswell) [reply doc/neps/nep-0029-deprecation_policy.rst]: In the guarantee for releases right after a python minor enters support I am counting on the back-compatibility between minor versions of Python such that versions of our projects that work on 3.N, should work out of the box with version 3.(N+1). 

This back-compatibility has not been perfect on the c-api front, but a combination of cython + cpython seems to be pretty back-compatible.

Also related, everyone needs to stop shipping cythonized .c files in sdists, but that is a different discussion.

We could hedge this with  "conditional on Python's back-compat" or change it to "we expect that there will be a non-empty set of supported Python versions by all actively maintained projects"

Comment (stefanv) [reply doc/neps/nep-0029-deprecation_policy.rst]: OK, I see where the confusion came in: I was thinking of *binary* distributions, and you are talking about source.  The source will be compatible, while the *binary* support may not be available.  The *binary* support depends on the team releasing rapidly, while the *source* support is there without any action on the developers' behalf.

With skimage, we ship Cythonized files that are used *only* when Cython is not available; problematic?

W.r.t. the "hedging", if we are talking about source that does not seem necessary; Python should not break API compatibility on minor versions, and the latter suggested phrase feels so pessimistic compared to the likely reality.

Comment (stefanv) [reply doc/neps/nep-0029-deprecation_policy.rst]: Resolve by ee9ba282886ed1c3a927abcfe21cf42242f5d579.

**Review by tacaswell — APPROVED** [review]

**Comment by tacaswell** [comment]
I think this is good to go!

**Comment by mattip** [comment]
Thanks @tacaswell

**Pull request merged**